aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -361,9 +412,23 @@ public void onResult(SignInResult result) {
 
             @Override
             public void onError(Exception error) {
-                onException.accept(
-                    new AuthException("Sign in failed", error, "See attached exception for more details")
-                );
+                if (error instanceof InvalidPasswordException) {
```

Comment:
A few thoughts on the error mapping strategy:

You could make a:
```java
final class CognitoErrorMapping {
    private CognitoErrorMapping() {}

    static AWSCognitoAuthException lookup(Exception exception) {
        if (exception instanceof ...) {
            return ...
        }
    }
}
```
and then centralize the mapping in a separate class. This would help keep the length of `AWSCognitoAuthPlugin` down, too.

Secondly, can you include the _entire_ `error` in the new exception's "cause" field, instead of just extracting the `getCause()`? Right now, I think you're removing/masking one layer in the exception trace. But better would be to simply _wrap_ the entire thing, into our types.